Temporary Images Resizing on Resolution Change (Part 2)

authenticate opened the pull request:
Hi,

Here is the second part of the other pull request.  This pull request contains the actual fix.

Let me know what you think or if you have any questions.  I added some helper functions to "image.h" to consolidate the math into one location.  I just put the functions in there for now.  I was not sure exactly where you would want them.

In addition, I was not able to test the changes in "menu.cpp".  I am not sure where in the game that section of code is called.  Maybe it is never called?

As a side note, when these pull requests get ready to merge, it might be easier to merge the other one first.  Then, I can rebase this pull request on top of it to make things easier.

Thanks.

Bertram25 commented:
Sorry, you will need to rebase and fix conflicts. But this shouldn't be that tough, especially since it's your changes.

authenticate commented:
It was not a problem.

I rebased and merged everything.

I also added a small commit to address an issue I caused with the explicit constructor changes.  I accidentally marked a copy constructor as explicit.  I think the original battle command code is correct.  Sorry about that.

I am out of town and won't be home for a couple of days.  I do not have access to compiler and build tools for a few days.  So, I never actually compiled the most recent changes or tested them.  Hopefully nothing is wrong. :D

If there is an issue, let me know.

Thanks.

Bertram25 commented:
> I also added a small commit to address an issue I caused with the explicit constructor changes. I accidentally marked a copy constructor as explicit. I think the original battle command code is correct. Sorry about that.

Np, I wasn't sure the best way to go while testing to I took the quick-road by quick fixing it. But thanks for taking care of it.

> I am out of town and won't be home for a couple of days. I do not have access to compiler and build tools for a few days. So, I never actually compiled the most recent changes nor tested them. Hopefully nothing is wrong. :D I just wanted to get this cleaned up so you were not waiting on me.

Eh eh. Np, everything went fine when I tested it. I'll merge now. Thanks a lot for the job done!! :D
